transformer_engine/pytorch/transformer.py

```diff
@@ -7,9 +7,9 @@
 import math
 import warnings
 from importlib.metadata import version
-from distutils.version import LooseVersion
```

Comment:
Switching from `LooseVersion` to `Version` may cause issues if we ever run into a non-standard version name:
```python
import flash_attn
if LooseVersion(version("flash_attn")) >= LooseVersion("1.2.3-myweirdfeature"):
    _enable_amazing_optimiztion = True
```
That said, it may be good to impose the extra discipline. Version strings ending in `-dev` are still valid.